### PR TITLE
Add background image to guake

### PR DESCRIPTION
**Pull request by mlouielu**
Impl #1603.

Impl detail please refer to `guake/utils.py:BackgroundImageManager.draw`

### CODE REVIEW

**Comment by mlouielu** [comment]
It should be used in any modern VTE version (it doesn't invoke any VTE methods), and anyone should able to use it without problem.

**Comment by gsemet** [comment]
Oh ok you don’t use any vte function. I guess this may take some cpu to run, can check if when disabled it really does not have any impact on perf

**Comment by mlouielu** [comment]
It now should cached when loading the same file (e.g. when restoring tabs).

**Comment by moonlighf** [comment]
i love this feature, but your branch lose this feature that search in current terminal

File: guake/data/org.guake.gschema.xml [hunk 1]
```diff
@@ -166,6 +166,16 @@
             <summary>Window vertical alignment.</summary>
             <description>Place Guake at: 0: top, 1: bottom</description>
         </key>
+        <key name="background-image-file" type="s">
+          <default>''</default>
```
Comment (la5942) [inline guake/data/org.guake.gschema.xml]: Indents are all off here

File: guake/utils.py [hunk 2]
```diff
@@ -20,12 +20,15 @@
...
 """
 import datetime
+import enum
 import logging
 import os
 import platform
 import subprocess
 import time
 
+import cairo
```
Comment (Davidy22) [inline guake/utils.py:30]: Does this need to be added to requirements.txt? Looks like a new dependency.

**Review by Davidy22 — COMMENTED** [review]
Tried it out and was sort of hoping for image transparency, but this is still nice and pretty without that. Would be nice if we used a file picker that showed thumbnails too. Both of those can get added in another commit though.

File: guake/utils.py (hunk 2 above)
Comment (gsemet) [reply guake/utils.py:30]: requirements.txt is maintained by pipenv

Comment (Davidy22) [reply guake/utils.py:30]: Oh alright, just @la5942's review stands then.

**Comment by Davidy22** [comment]
In addition to polishing off that whitespace objection from @la5942, also going to need to grab the recent CI changes and make sure this passes the build and then we can merge.

**Comment by Davidy22** [comment]
Went and committed the little whitespace edit myself, just some merges and then a passthrough of black left and this should be good to go.

**Review by Davidy22 — APPROVED** [review]
Went and did the rebase myself. Still works, passes checks, was already approved before so merging.

**Pull request merged**